crates/holochain/src/core/workflow/sys_validation_workflow.rs

```diff
@@ -159,48 +162,86 @@ pub async fn sys_validation_workflow(
+    let missing_dependencies = current_validation_dependencies
         .lock()
         .expect("poisoned")
-        .get_missing_hashes();
-    let num_fetched: usize = futures::stream::iter(missing_action_hashes.into_iter().map(|hash| {
-        let network_cascade = network_cascade.clone();
-        let current_validation_dependencies = current_validation_dependencies.clone();
-        async move {
-            match network_cascade
-                .retrieve_action(hash, Default::default())
-                .await
-            {
-                Ok(Some((action, source))) => {
-                    let mut deps = current_validation_dependencies.lock().expect("poisoned");
-
-                    // If the source was local then that means some other fetch has put this action into the cache,
-                    // that's fine we'll just grab it here.
-                    if deps.insert(action, source) {
-                        1
-                    } else {
-                        0
+        .get_missing_dependencies();
+    let num_fetched: usize =
+        futures::stream::iter(missing_dependencies.into_iter().map(|(hash, dep_type)| {
+            let network_cascade = network_cascade.clone();
+            let current_validation_dependencies = current_validation_dependencies.clone();
+            async move {
+                match dep_type {
+                    ValidationDependencyType::Action => {
+                        match network_cascade
+                            .retrieve_action(hash, Default::default())
+                            .await
+                        {
+                            Ok(Some((action, source))) => {
+                                let mut deps =
+                                    current_validation_dependencies.lock().expect("poisoned");
+
+                                if deps.insert_action(action, source) {
+                                    1
+                                } else {
+                                    0
+                                }
+                            }
+                            Ok(None) => {
+                                // This is fine, we didn't find it on the network, so we'll have to try again.
+                                // TODO This will hit the network again fairly quickly if sys validation is triggered again soon.
+                                //      It would be more efficient to wait a bit before trying again.
+                                0
+                            }
+                            Err(err) => {
+                                tracing::error!(?err, "Error fetching missing dependency");
+                                0
+                            }
+                        }
+                    }
+                    ValidationDependencyType::Warranted(chain_op_type) => {
+                        println!("Fetching warranted dependency from network: {}", hash);
```

Comment:
Are these `println!`s intended to be kept? There's one a few lines below as well.

Reply:
Should have been removed, they were for debugging. Thanks for picking that up.